### ISSUE TITLE
Class 'Magento\Setup\Application' not found

### ISSUE THREAD

**Issue by sureshpareek** [magento/magento2#12708]
Today I Upgrade my Magento from 2.1.7 to 2.2.2. In this process, I got one error that I cannot resolve. and this comes with every command. if I run command php bin/Magento -v then also this problem comes up.

> Fatal error: Uncaught Error: Class 'Magento\Setup\Application' not found in /home/epic/public_html/vendor/magento/framework/Console/Cli.php:71
Stack trace:
#0 /home/epic/public_html/bin/magento(21): Magento\Framework\Console\Cli->__construct('Magento CLI')
#1 {main}
  thrown in /home/epic/public_html/vendor/magento/framework/Console/Cli.php on line 71

Now I do not know why this problem is coming. I show everything below is my /home/epic/public_html/vendor/magento/framework/Console/Cli.php file code when problem coming

> public function __construct($name = 'UNKNOWN', $version = 'UNKNOWN')
    {
        try {
            $configuration = require BP . '/setup/config/application.config.php';
            $bootstrapApplication = new Application();
            $application = $bootstrapApplication->bootstrap($configuration);
            $this->serviceManager = $application->getServiceManager();
           $this->assertCompilerPreparation();
            $this->initObjectManager();
            $this->assertGenerationPermissions();
        } catch (\Exception $exception) {
            $output = new \Symfony\Component\Console\Output\ConsoleOutput();
            $output->writeln(
                '<error>' . $exception->getMessage() . '</error>'
            );
           exit(static::RETURN_FAILURE);
        }
       if ($version == 'UNKNOWN') {
            $directoryList = new DirectoryList(BP);
            $composerJsonFinder = new ComposerJsonFinder($directoryList);
            $productMetadata = new ProductMetadata($composerJsonFinder);
            $version = $productMetadata->getVersion();
        }
       parent::__construct($name, $version);
    }

**Comment by sureshpareek** [comment]
any update please?

**Comment by sureshpareek** [comment]
any update, please?

**Comment by wclansar** [comment]
I am also getting same problem after upgrade in magento 2.2.2

PHP Fatal error:  Uncaught Error: Class 'Magento\Setup\Application' not found in /var/www/html/vytronix/site/magento/vendor/magento/framework/Console/Cli.php:71
Stack trace:
#0 /var/www/html/vytronix/site/magento/bin/magento(22): Magento\Framework\Console\Cli->__construct('Magento CLI')
#1 {main}
  thrown in /var/www/html/vytronix/site/magento/vendor/magento/framework/Console/Cli.php on line 71

**Comment by akilirw** [comment]
When I updated via composer, it deleted some files from the `setup` folder. At first my error was: 

<pre>$ php bin/magento 
PHP Warning:  Uncaught Exception: Warning: require(/vagrant/setup/config/application.config.php): failed to open stream: No such file or directory in /vagrant/vendor/magento/framework/Console/Cli.php on line 70 in /vagrant/vendor/magento/framework/App/ErrorHandler.php:61
Stack trace:
#0 /vagrant/vendor/magento/framework/Console/Cli.php(70): Magento\Framework\App\ErrorHandler->handler(2, 'require(/vagran...', '/vagrant/vendor...', 70, Array)
#1 /vagrant/vendor/magento/framework/Console/Cli.php(70): require()
#2 /vagrant/bin/magento(22): Magento\Framework\Console\Cli->__construct('Magento CLI')
#3 {main}
  thrown in /vagrant/vendor/magento/framework/App/ErrorHandler.php on line 61
PHP Fatal error:  Magento\Framework\Console\Cli::__construct(): Failed opening required '/vagrant/setup/config/application.config.php' (include_path='/vagrant/generated/code:/vagrant/vendor/phpunit/php-file-iterator:/vagrant/vendor/phpunit/phpunit:/vagrant/vendor/symfony/yaml:/vagrant/vendor/magento/zendframework1/library:.:/usr/share/pear:/usr/share/php') in /vagrant/vendor/magento/framework/Console/Cli.php on line 70</pre>

Which I realized was due to `/setup/config/application.config.php` missing. So I just did a `git checkout setup` from my repo to retrieve that file and that's when I saw the error message that you are receiving:

<pre>$ php bin/magento 
PHP Fatal error:  Uncaught Error: Class 'Magento\Setup\Application' not found in /vagrant/vendor/magento/framework/Console/Cli.php:71
Stack trace:
#0 /vagrant/bin/magento(22): Magento\Framework\Console\Cli->__construct('Magento CLI')
#1 {main}
  thrown in /vagrant/vendor/magento/framework/Console/Cli.php on line 71</pre>

Which is because I was using the setup folder from 2.1.7 after the `git checkout`

After manually replacing the setup directory with the the one from `Magento-EE-2.2.2-2017-12-11-08-57-11.tar.gz` I am able to run Magento CLI commands. Though this isn't a fix, just a bandaid. The `setup` directory should update properly with composer. Not sure if this will help your situation, but it worked for me.

**tl;dr**:
For some reason it seems as though upgrading from 2.1.7 to 2.2.2 via composer didn't properly update the `setup` directory and I ended up manually adding it by downloading the 2.2.2 tar from my Magento commerce account. At least this is what happened in my case.

**Comment by coachdesign** [comment]
Have you solved in any way? I have the same problem.

**Comment by georgeschiopu** [comment]
@sureshpareek @wclansar Have the same problem with upgading from `2.1.7` to `2.2.2` - what PHP version are you guys having? We tried it on `7.0`, `7.1` (which should be fully compatible, according to http://devdocs.magento.com/guides/v2.3/install-gde/system-requirements-tech.html )

Got anywhere with it?

**Comment by sureshpareek** [comment]
I still have problem in one of my website. we are using php 7.1. but also checked to 7.0 and other 

----- Original Message -----

**Comment by wclansar** [comment]
Hello @georgeschiopu 
yes it has fixed no more issue.
Thanks

**Comment by riksof-zzlalani** [comment]
@wclansar How was it fixed? I have the same issue in my magento 2.1.11 when I tried to install a paid plugin, they have this tutorial http://www.codazon.com/document/fastest/magento2/index.html#install-theme 

and in the index.php file of themesetup it throws

Fatal error: Uncaught Error: Class 'Magento\Setup\Application' not found in /var/www/html/themesetup/index.php:33 Stack trace: #0 {main} thrown in /var/www/html/themesetup/index.php on line 33

I have php 7.0.25 and apache 2.4.27, I have also contacted codazon support but they are too slow to reply,

**Comment by georgeschiopu** [comment]
@riksof-zzlalani don't know if it's the same for `2.1` but for `2.2.2` but for me there was a problem with some of the files in the `/setup/` dir being *gitignored*. Check the integrity for that dir and compare it with a "vanilla" installation that you know it's working. Hope it helps.

**Comment by magento-engcom-team** [comment]
Hi @sureshpareek 
Thank you for your report.
We can not reproduce the issue as you describe.
May be you helped the last comment from @georgeschiopu ?

**Comment by magento-engcom-team** [comment]
@sureshpareek we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.

**Comment by magento-engcom-team** [comment]
@sureshpareek, thank you for your report.
This seems to be correct Magento behavior. Please refer to the [Community Forums](https://community.magento.com/) or the [Magento Stack Exchange](https://magento.stackexchange.com/) site for advice or general discussion about this.
Otherwise you may submit Pull Request with the suggested changes.

**Comment by SamRogal** [comment]
I also received following error message after upgrading from 2.1.9 to 2.2.3
```
20-Apr-2018 18:12:26 UTC] PHP Fatal error:  Uncaught Error: Class 'Magento\Setup\Application' not found in /home/smkwm2/Magento/vendor/magento/framework/Console/Cli.php:71
```

To correct for the issue after `composer update` had successfully completed once I ran 
```
# cd to you magento directory
rm -rf vendor/*
composer clearcache
composer install  -o --prefer-dist --no-progress --profile --ignore-platform-reqs
```

In my case the composer cache seemed to be holding on to something that broke generating the autoload file correctly.
